src/core/psi_container.jl

```diff
@@ -391,28 +374,73 @@ function add_cons_container!(psi_container::PSIContainer, cons_name::Symbol, axs
     return container
 end
 
-function get_parameter(psi_container::PSIContainer, ref::UpdateRef)
-    parameter = get(psi_container.parameters, ref, nothing)
-    if isnothing(parameter)
-        @error "$ref is not stored" sort!(get_parameter_refs(psi_container))
-        throw(IS.InvalidValue("parameter $ref is not stored"))
+function get_parameter_names(psi_container::PSIContainer)
+    return collect(keys(psi_container.parameters))
+end
+
+function get_parameter_container(psi_container::PSIContainer, name::AbstractString)
+    return get_parameter_container(psi_container, Symbol(name))
+end
+
+function get_parameter_container(psi_container::PSIContainer, name::Symbol)
+    container = get(psi_container.parameters, name, nothing)
+    if isnothing(container)
+        @error "$name is not stored" sort!(get_parameter_names(psi_container))
+        throw(IS.InvalidValue("parameter $name is not stored"))
     end
         
-    return parameter
+    return container
 end
 
-function get_parameter_refs(psi_container::PSIContainer)
-    return collect(keys(psi_container.parameters))
+function get_parameter_container(
+    psi_container::PSIContainer,
+    name::Symbol,
+    ::Type{T},
+) where T<:PSY.Component
+    return get_parameter_container(psi_container, _encode_for_jump(T, name))
 end
 
-function assign_parameter(psi_container::PSIContainer, ref::UpdateRef, value)
-    @debug "set_parameter" ref
-    psi_container.parameters[ref] = value
+function get_parameter_container(psi_container::PSIContainer, ref::UpdateRef)
+    return get_parameter_container(psi_container, ref.access_ref)
+end
+
+function get_parameter_array(psi_container::PSIContainer, ref)
```

Comment:
It's possible that you will want more implementations of this function. I didn't want to repeat each case of `get_parameter_container`, but am open.